pkg/dbmate/db.go

```diff
@@ -233,7 +234,7 @@ func (db *DB) LoadSchema() error {
 
 	_, err = os.Stat(db.SchemaFile)
 	if err != nil {
-		return nil
+		return ErrSchemaNotFound
 	}
```

Comment:
Personally, I would prefer you just return `err` here, because you are assuming that `os.IsNotExist(err)` is true here with the error message you're returning, but that may not necessarily be the case.

Unfortunately, it's hard to automate a test around other situations where `os.Stat()` may return an error because if you're [using a Mac to run Docker, then permission handling on volume mounts makes this difficult](https://github.com/docker/for-mac/issues/2657).  We can use a container-native filesystem mountpoint like `/tmp`, though:

```sh
$ docker compose run --rm dev /bin/sh

$ mkdir /tmp/inaccessible
$ touch /tmp/inaccessible/unreadable
$ chown -R 0:0 /tmp/inaccessible
$ chmod 0 /tmp/inaccessible

$ su nobody -s /bin/sh -c "dist/dbmate --url $POSTGRES_TEST_URL --schema-file /tmp/inaccessible/unreadable load"

## Output:

Error: stat /tmp/inaccessible/unreadable: permission denied
```

That's what `dbmate` will output if you just return `err` up to the caller.  The built-in error messages are concise and specific and clear:

```sh
$ su nobody -s /bin/sh -c "dist/dbmate --url $POSTGRES_TEST_URL --schema-file /tmp/nonexistent load"
Error: stat /tmp/nonexistent: no such file or directory
```

Reply:
I agree with you, it will be clearer for the user in this case than just general error.

And special thanks for explaining how to simulate it in docker on Mac.